tuist/tuist
# Add core data models to the sources build phase

Resolves https://github.com/tuist/tuist/issues/1539

### Short description 📝

- The core data models were previously added to the resources build phase
- Those models need to be included in the sources build phase so they get compiled

### Solution 📦

- Update the build phase generator to include core data models in the sources build phase instead
- Update the fixture to reference some of the core data generated models (e.g. `User`) to help test and validate the correct behaviour

### Implementation 👩‍💻👨‍💻

- [x] Update build phase generator
- [x] Validate correctness of the change 
  _(should core data models be in Resources or Sources or does it depend on another factor?)_


### Test Plan 🛠

- Run `swift run tuist generate` within `fixtures/ios_app_with_coredata/`
- Open the generated project
- Verify the core data models are within the Sources build phase
- Verify the app builds successfully without any errors
